Loop/Views/Presets/PresetRangeEditor.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  EditPressRangeView.swift
+//  Loop
+//
+//  Created by Pete Schwamb on 12/17/24.
+//  Copyright © 2024 LoopKit Authors. All rights reserved.
+//
+import SwiftUI
+import LoopAlgorithm
+import LoopKit
+import LoopKitUI
+
+struct PresetRangeEditor: View {
+    @Environment(\.dismiss) private var dismiss
+    @Environment(\.guidanceColors) private var guidanceColors
+    @EnvironmentObject private var displayGlucosePreference: DisplayGlucosePreference
+
+    @Binding var range: ClosedRange<LoopQuantity>?
+    var guardrail: Guardrail<LoopQuantity>
+    private var scheduledRange: ClosedRange<LoopQuantity>
+    private var allowsScheduledRange: Bool
+    private var isPreMeal: Bool
+
+    init(range: Binding<ClosedRange<LoopQuantity>?>, guardrail: Guardrail<LoopQuantity>, scheduledRange: ClosedRange<LoopQuantity>, allowsScheduledRange: Bool = true, isPreMeal: Bool) {
+        self._range = range
+        self.guardrail = guardrail
+        self.scheduledRange = scheduledRange
+        self.allowsScheduledRange = allowsScheduledRange
+        self.isPreMeal = isPreMeal
+    }
+
+    var displayedRange: ClosedRange<LoopQuantity> {
+        return range ?? scheduledRange
+    }
+
+    func boundText(for bound: LoopQuantity) -> Text {
+        let color = guardrail.color(for: bound, guidanceColors: guidanceColors)
+        let text = displayGlucosePreference.format(bound, includeUnit: false)
+        switch guardrail.classification(for: bound) {
+        case .withinRecommendedRange:
+            return Text(text)
+                .foregroundColor(range == nil ? .secondary : .accentColor)
+                .font(.system(size: 42, weight: .semibold))
+        case .outsideRecommendedRange:
+            return (
+                Text(Image(systemName: "exclamationmark.triangle.fill"))
+                    .font(.system(size: 29, weight: .regular))
+                    .baselineOffset(3.0)
+                    .foregroundColor(color) +
+                Text(text)
+                    .foregroundColor(color)
+                    .font(.system(size: 42, weight: .semibold))
+                )
```

Comment:
All of the manual `.font`s in this file wont scale with dynamic type. Is that alright?

Reply:
Yeah I think it's ok here, for big fonts.